openverse_catalog/dags/data_refresh/refresh_view_data_task_factory.py

```diff
@@ -0,0 +1,39 @@
+"""
+# Refresh Materialized View Task Factory
+This file generates a Task that refreshes the materialized view for a
+given media type, using a factory function.
+
+The task refreshes the materialized view, but not the underlying tables. This
+means that the only effect is to add or update data (including popularity data)
+for records which have been ingested since the last time the view was
+refreshed.
+
+This should be run every time before a data refresh is triggered.
+"""
+import os
+
+from airflow.utils.trigger_rule import TriggerRule
+from common.popularity import operators
+
+
+DB_CONN_ID = os.getenv("OPENLEDGER_CONN_ID", "postgres_openledger_testing")
+
+
+def create_refresh_view_data_task(media_type: str):
```

Comment:
Just a thought - we're referencing `media_type` more and more, which I love! Maybe we should make it a [`typing.Literal` with `image`/`audio` options](https://docs.python.org/3/library/typing.html#typing.Literal) for clarity?

Reply:
I like this idea! I'm not sure what the cleanest way is to make it extensible for additional model, though. I could define a `MEDIA_TYPES_TYPE` (😬) in `constants.py` for example. It looks like you can't dynamically create a Literal from an array, so we can't do something like `typing.Literal[MEDIA_TYPES]`. We'd need to make sure the type itself is updated when new models are added.

Reply:
Hmm, that sounds okay to me? I mean, we have media types defined in both the [API](https://github.com/WordPress/openverse-api/blob/270b086095ba125fa26e04bf27dff3ab6994e025/api/catalog/api/constants/media_types.py) and the [data refresh server](https://github.com/WordPress/openverse-api/blob/270b086095ba125fa26e04bf27dff3ab6994e025/ingestion_server/ingestion_server/constants/media_types.py) 🤷🏼‍♀️

Reply:
Fair enough :) I'm just annoyed something like `Literal[MEDIA_TYPES]` doesn't seem to be possible, considering we already have `MEDIA_TYPES` defined 😆

Reply:
Apparently you can do this, sort of, you just have to go the other way around: https://stackoverflow.com/a/64522240